src/js/index.js

```diff
@@ -52,4 +38,23 @@ import { renderAllPrograms } from './render/render';
     console.warn(`Please set process.env.TYPE properly in package.json, current value: ${process.env.TYPE}`);
   }
   }
-})();
\ No newline at end of file
+})();
+
+function initShareAndRenderPrograms(programPath) {
```

Comment:
javascript code for share.html and reder.html is almost the same. except program path. So moved code to new function and calling it twice.